protocol/contracts/reserve/src/contract.rs

```diff
@@ -0,0 +1,118 @@
+            SingleUserAccess::new(
+                deps.storage.deref_mut(),
+                crate::access_control::LEASE_CODE_ADMIN_KEY,
+            )
+            .grant_to(&new_reserve.lease_code_admin)
+            .map_err(Into::into)
+        })
+        .and_then(|()| versioning::initialize(deps.storage, CONTRACT_VERSION).map_err(Into::into))
+        .and_then(|()| Config::from(new_reserve).store(deps.storage))
+        .map(|()| response::empty_response())
+        .or_else(|err| platform_error::log(err, deps.api))
+}
+
+#[entry_point]
+pub fn migrate(deps: DepsMut<'_>, __env: Env, _msg: MigrateMsg) -> ContractResult<CwResponse> {
+    versioning::update_software(deps.storage, CONTRACT_VERSION, Into::into)
+        .and_then(response::response)
+        .or_else(|err| platform_error::log(err, deps.api))
+}
+
+#[entry_point]
+pub fn execute(
+    mut deps: DepsMut<'_>,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> ContractResult<CwResponse> {
+    match msg {
+        ExecuteMsg::NewLeaseCode {
+            code_id: new_code_id,
+        } => {
+            SingleUserAccess::new(
+                deps.storage.deref_mut(),
+                crate::access_control::LEASE_CODE_ADMIN_KEY,
+            )
+            .check(&info.sender)?;
+
+            Config::update_lease_code(deps.storage, new_code_id.into())
+                .map(|()| PlatformResponse::default())
+        }
+        ExecuteMsg::CoverLiquidationLosses { amount } => {
+            let lease = info.sender;
+
+            Config::load(deps.storage)
+                .and_then(|config| {
+                    contract::validate_code_id(deps.querier, &lease, config.lease_code_id())
+                        .map_err(ContractError::from)
+                })
+                .and_then(|()| amount.try_into().map_err(Into::into))
+                .and_then(|amount: Coin<LpnCurrency>| {
+                    let mut bank = bank::account(&env.contract.address, deps.querier);
+                    bank.balance::<LpnCurrency>()
+                        .map_err(Into::into)
+                        .and_then(|balance| {
+                            if balance < amount {
+                                Err(ContractError::InsufficientBalance)
+                            } else {
+                                bank.send(amount, &lease);
+                                let msg: Batch = bank.into();
+
+                                Ok(msg.into())
+                            }
+                        })
+                })
+        }
+    }
+    .map(response::response_only_messages)
+    .or_else(|err| platform_error::log(err, deps.api))
+}
+
+#[entry_point]
+pub fn query(deps: Deps<'_>, _env: Env, msg: QueryMsg) -> ContractResult<Binary> {
+    match msg {
+        QueryMsg::Config() => Config::load(deps.storage)
+            .map( Into::<ConfigResponse>::into)
+            .and_then(|config| cosmwasm_std::to_json_binary(&config).map_err(Into::into))
```

Comment:
[Suggestion] I'd prefer using `sdk::cosmwasm_std` personally. In case we ever remove the direct dependency, it would be easier to handle.

Reply:
I follow the Rust calling conventions for functions